presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/client/ElasticsearchClient.java

```diff
@@ -502,6 +503,9 @@ public IndexMetadata getIndexMetadata(String index)
             catch (IOException e) {
                 throw new PrestoException(ELASTICSEARCH_INVALID_RESPONSE, e);
             }
+            catch (NoSuchElementException e) {
+                throw new PrestoException(ELASTICSEARCH_QUERY_FAILURE, "SELECT * not allowed from relation that has no columns");
```

Comment:
How can we be sure this exception is thrown only for this scenario?

Reply:
> if (!mappings.has("properties")) {
>                     mappings = mappings.elements().next();
>                 }


is responsible for iterating over the mappings (i.e., columns) within the index. If no columns were specified during the index (table) creation, the code attempts to access the next element, which results in a `NoSuchElementException` because there are no elements to iterate over.

Since this exception is not explicitly handled, it causes an internal error to be thrown. The entire try block is dedicated to handling column-related logic. Therefore, as long as columns are present, the code will proceed without any issues, and the `NoSuchElementException` scenario will not arise beyond this point in the block.

This exception is specific to the situation where no columns are defined in the index, which is why it occurs at this stage. If columns are correctly defined, the flow will continue normally without encountering this error.

Reply:
How can we be sure only `select *` causes this scenario.

Reply:
Yes there is a possibility with other queries as well. I will change the error message to generic to all.